### PR TITLE
Improving balancing and bert approaches

### PR DESCRIPTION
**Pull request by Smu-Tan**

### CODE REVIEW

**Review by svenvanderburg — COMMENTED** [review]
Nice @Smu-Tan!

Some comments:

- [ ] If you have so many imports it is good to organise them a bit. See [this link](https://www.python.org/dev/peps/pep-0008/#:~:text=Imports%20are%20always%20put%20at,Related%20third%20party%20imports.) for PEP8 convention, and in addition most people like putting them in alphabetical order.
- [ ] For the notebook to work properly you need to be in root folder right? So maybe add a cell with `cd ..` (so we go the the parent directory of the notebooks folder, which is the root).
- [ ]  Instead of your balancing function you can use [scikitlearn resample](https://scikit-learn.org/stable/modules/generated/sklearn.utils.resample.html)
- [ ] It would be nice if at the top of the notebook you can add a summary/goal of the notebook
- [ ] Maybe you can rename the notebook so that it matches the convention of the rest of the notebooks and give it a bit more informative name
- [ ] Maybe you can add a bit more documentation on the approach with BERT hidden states and then using a linear regression model? Is this the recommended way of doing classification with BERT? Why did not you not use [BertForSequenceClassification](https://huggingface.co/transformers/model_doc/bert.html#transformers.BertForSequenceClassification)?
- [ ] I think it makes more sense to first remove missing data, and then do balancing. As I said before, we should do the missing data removal in the data preprocessing step. Can you create a new issue for that?
- [ ] The scikitlearn `classification_report` is a useful function! I forgot about it, thanks for me letting me remember :)
- [ ] The results look promising, but it could very well be that this is because you test the model on the training data. (i.e. the model could have just remembered the labels instead of learning the task). How long did the training take? If it's less than let's say 5 minutes you could use the cross-validation like we use in the other notebook. If it takes longer we can maybe do a 60%-20%-20% train-val-test split (I think it would be good to make stratified splits based on the labels, use the `stratify` argument of `train_test_split`). Then we will see whether the results hold yes or no!
